lib/MadNLPHSL/src/ma27.jl

```diff
@@ -48,7 +48,7 @@ mutable struct Solver <: AbstractLinearSolver
     logger::Logger
 end
 
-ma27ad!(n::Cint,nz::Cint,I::StrideOneVector{Cint},J::StrideOneVector{Cint},
+ma27ad!(n::Cint,nz::Cint,I::AbstractVector{Cint},J::AbstractVector{Cint},
```

Comment:
I agree that `StrideOneVector` is not ideal, but the motivation for using it was to make sure that when calling linear solvers, we don't accidentally pass AbstractVectors with non-uniform memory stride (in particular, we want it to be always 1 byte). My concern with this change is that we don't have anything that prevents `ma27ad!` to accept a pointer to a non-stride-one-array. Of course, we're using these only internally within MadNLP, so shouldn't be a problem if we write the code carefully in the future, but it raises a bit of concern to me. Any thoughts on this @frapac?

Reply:
To be honest, I have exactly the same concern as you. I thought about this the last few days, and I may have a potential workaround. How about adding a safety check at the beginning of the function: 
```julia
@assert strides(I) == (1, )
```